util/test/Test/Pos/Util/Base16.hs

```diff
@@ -58,3 +66,32 @@ chunkBS n xs = case LB.uncons xs of
     Just _  ->
         let (taken, dropped) = LB.splitAt n xs
         in  taken : chunkBS n dropped
+
+----------------------------------------------------------------------------
+-- Decoding
+----------------------------------------------------------------------------
+
+-- | Decode a given ByteString which was originally encoded using 'encode' or
+-- 'encodeWithIndex'.
+decode :: LB.ByteString -> LB.ByteString
+decode bs
+    -- No complex parsing is required for data whose length is <= 32.
+    | LB.length bs <= lineWrapLength = fst $ B16.decode bs
+    | otherwise = case PLB.eitherResult $ PLB.parse decodeParser bs of
+        Left l  -> error $ fromString l
+        Right r -> fst $ B16.decode $ LB.fromStrict $ BC.concat r
```

Comment:
@erikd Should I avoid using lazy ByteStrings unless absolutely necessary? attoparsec seems to use strict ByteStrings throughout its codebase and this forced me to use "fromStrict".